biosys/apps/main/api/views.py

```diff
@@ -55,7 +44,7 @@ def has_permission(self, request, view):
         return \
             request.method in SAFE_METHODS \
             or is_admin(user) \
-            or (hasattr(view, 'dataset') and view.dataset.is_custodian(user))
+            or (hasattr(view, 'dataset') and view.dataset and view.dataset.is_custodian(user))
```

Comment:
Only change if you want, but this could be a good opportunity to use Python's `all` and `any` keywords.